spark-connector/spark-common/src/test/java/org/apache/gravitino/spark/connector/integration/test/sql/SparkQueryRunner.java

```diff
@@ -0,0 +1,351 @@
+import org.apache.gravitino.integration.test.container.HiveContainer;
+import org.apache.gravitino.integration.test.util.AbstractIT;
+import org.apache.gravitino.spark.connector.GravitinoSparkConfig;
+import org.apache.gravitino.spark.connector.iceberg.IcebergPropertiesConstants;
+import org.apache.gravitino.spark.connector.plugin.GravitinoSparkPlugin;
+import org.apache.spark.sql.SparkSession;
+import org.junit.jupiter.api.Assertions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Run and check the correctness of the SparkSQLs */
+public class SparkQueryRunner {
+
+  private static final Logger LOG = LoggerFactory.getLogger(SparkSQLRegressionTest.class);
+  private static final String HIVE_CATALOG_NAME = "hive";
+  private static final String ICEBERG_CATALOG_NAME = "iceberg";
+
+  private SparkSession sparkSession;
+  private String gravitinoUri;
+  private String metalakeName;
+  private String warehouse;
+  private boolean regenerateGoldenFiles;
+  // catalogType -> catalogName
+  private Map<CatalogType, String> catalogs = new HashMap<>();
+  private boolean isSetupGravitinoEnv;
+  private String dataDir;
+
+  private static final ContainerSuite containerSuite = ContainerSuite.getInstance();
+
+  public SparkQueryRunner(SparkTestConfig sparkTestConfig) {
+    this.regenerateGoldenFiles = sparkTestConfig.generateGoldenFiles();
+    this.isSetupGravitinoEnv = sparkTestConfig.isSetupGravitinoEnv();
+    String baseDir = sparkTestConfig.getBaseDir();
+    // translate to file:///xx/xx
+    this.dataDir = Paths.get(baseDir, "data").toUri().toString();
+    this.metalakeName = sparkTestConfig.getMetalakeName();
+    if (isSetupGravitinoEnv) {
+      try {
+        setupGravitinoEnv();
+      } catch (Exception e) {
+        throw new RuntimeException(e);
+      }
+    } else {
+      this.gravitinoUri = sparkTestConfig.getGravitinoUri();
+      this.warehouse = sparkTestConfig.getWarehouseLocation();
+    }
+    initSparkEnv();
+
+    catalogs.put(CatalogType.HIVE, HIVE_CATALOG_NAME);
+    catalogs.put(CatalogType.ICEBERG, ICEBERG_CATALOG_NAME);
+    catalogs.put(CatalogType.UNKNOWN, HIVE_CATALOG_NAME);
+  }
+
+  public void runQuery(TestCaseGroup sqlTestCaseGroup) throws IOException {
+    useCatalog(sqlTestCaseGroup.getCatalogType());
+    try {
+      if (sqlTestCaseGroup.prepareFile != null) {
+        runQuery(sqlTestCaseGroup.prepareFile);
+      }
+      sqlTestCaseGroup.testCases.forEach(
+          testCase -> {
+            try {
+              runTestCase(testCase);
+            } catch (IOException e) {
+              throw new RuntimeException(e);
+            }
+          });
+    } finally {
+      if (sqlTestCaseGroup.cleanupFile != null) {
+        runQuery(sqlTestCaseGroup.cleanupFile);
+      }
+    }
+  }
+
+  public void close() throws Exception {
+    if (sparkSession != null) {
+      sparkSession.close();
+      sparkSession = null;
+    }
+    if (isSetupGravitinoEnv) {
```

Comment:
This variable name is not weird, better change to a new name, like `isGravitinoEnvSetUp`.

Reply:
done